### PR TITLE
ARTEMIS-548 Stomp durable sub unsubscrbe

### PR DESCRIPTION
**Pull request by jbertram**
Implement ability for Stomp clients to unsubscribe durable
subscriptions.

### CODE REVIEW

File: artemis-protocols/artemis-stomp-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/stomp/StompProtocolManager.java [hunk 1]
```diff
@@ -393,7 +393,8 @@ public void unsubscribe(StompConnection connection,
                            String subscriptionID,
                            String durableSubscriberName) throws Exception {
       StompSession stompSession = getSession(connection);
-      boolean unsubscribed = stompSession.unsubscribe(subscriptionID, durableSubscriberName);
+      String clientID = (connection.getClientID() != null) ? connection.getClientID() : null;
```
Comment (johnament) [inline artemis-protocols/artemis-stomp-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/stomp/StompProtocolManager.java]: null check looks redundant, its always the value of `connection.getClientID()`

**Comment by mtaylor** [comment]
@jbertram It looks like subscribe in STOMP is broker completely.  It might be worth looking at it before we rework unsubscribe.  The subscription queues are not unique across connections.

**Comment by jbertram** [comment]
There's a handful of Stomp subscriber tests running without issues.  Can you clarify a bit on what is broken.  If you can outline a specific use-case then I can write a test for it.

**Comment by mtaylor** [comment]
From looking at the code I believe normal subscription queues are not encapsulated via connection and instead are global.  Specifically, the subscription queue name is based on the subscription id. This is only unique in the context of the connection.  Another connection could specify the same subscription id for something else completely.

This could result in: 

1. Client 2 receives messages from the wrong address.
2. Client 1 unsubscribe deletes Client 2 subscription queue.
3. When both addresses are the same, p2p semantics instead of pub/sub.

**Comment by mtaylor** [comment]
@jbertram I noticed this when looking at unsubscribe: https://github.com/apache/activemq-artemis/blob/master/artemis-protocols/artemis-stomp-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/stomp/StompSession.java#L323

It could just be unsubscribe that's behaving funny...

**Pull request closed without merging**